core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -176,33 +176,20 @@ trait Task extends Dependable {
     *
     * @return the "other" task so that chains of calls like a ==> b ==> c work correctly
     */
-  override def ==>(other: Task) : other.type = {
-    other.dependsOnTasks += this
-    this.dependedOnByTasks += other
-    other
-  }
+  override def ==>(other: Dependable) : Dependable = {
+    other.toTasks.foreach(t => {
+      t.dependsOnTasks += this
+      this.dependedOnByTasks += t
+    })
 
-  override def ==>(other: MultiTask) : other.type = {
-    other.tasks.foreach(this ==> _)
-    other
+    new DependencyChain(this, other)
```

Comment:
I've modified things so that `==>` is implemented and final in `Dependable` so that the behaviour is guaranteed, and created a new method `addDependent` that is abstract and must be implemented by sub-classes to handle any side-effects.